### PR TITLE
Optional header

### PR DESCRIPTION
**Pull request by iisa**
Makes header optional for this component, does not draw by default.

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -72,15 +72,31 @@
...
       id: 'share',
+      title: 'Share this item',
       component: html`<ia-sharing-options
         identifier="goody"
         type="book"
         creator="Welsh, Charles"
         description="The history of Little Goody Two-Shoes : otherwise called Mrs. Margery Two-Shoes ... [1766 edition]"
-        baseHost="www-shaner.archive.org"
-      ></ia-sharing-options>`
+        baseHost="www-shaner.arc,hive.org"
```
Comment (shaneriley) [inline demo/index.html]: You've got a comment in the attribute value that should be removed.

File: package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "lit-html": "^1.1.2"
   },
   "devDependencies": {
-    "@internetarchive/ia-menu-slider": "0.1.0-alpha.1b6f614",
+    "@internetarchive/ia-menu-slider": "0.0.3-alpha.2",
```
Comment (shaneriley) [inline package.json]: Why did this version decrease?

File: src/ia-sharing-options.js [hunk 3]
```diff
@@ -104,11 +106,14 @@ export class IASharingOptions extends LitElement {
     this.embedOptionsVisible = !this.embedOptionsVisible;
   }
 
+  get header() {
+    const header = html`<header><h3>Share this ${this.type}</h3></header>`;
```
Comment (shaneriley) [inline src/ia-sharing-options.js:110]: I thought the header was being removed from individual submenu components and instead rendered by the menu slider component? Rendering this header within a shadowDOM like this means the close button and any other buttons meant to sit alongside the header have the potential to be overlapped by this header element.

File: src/ia-sharing-options.js [hunk 4]
```diff
@@ -33,6 +34,7 @@ export class IASharingOptions extends LitElement {
       identifier: { type: String },
       sharingOptions: { type: Array },
       type: { type: String },
+      renderHeader: { type: Boolean },
```
Comment (shaneriley) [inline src/ia-sharing-options.js:37]: I don't recall the use case for disabling the rendering of the header for submenu items. Is that how we're deferring header rendering to the parent menu slider?

**Review by shaneriley — CHANGES_REQUESTED** [review]

File: demo/index.html (hunk 1 above)
Comment (iisa) [reply demo/index.html]: ha! I thought that was something clever you put in 😅

File: src/ia-sharing-options.js (hunk 4 above)
Comment (shaneriley) [reply src/ia-sharing-options.js:37]: Isa confirmed it's to allow the parent menu slider to render the header instead.

File: demo/index.html (hunk 1 above)
Comment (iisa) [inline demo/index.html]:
```suggestion
        baseHost="archive.org"
```

File: demo/index.html [hunk 5]
```diff
@@ -72,15 +72,31 @@
...
+        <ia-icon-share></ia-icon-share>
+      `,
+      label: 'Share this item with header',
+      id: 'share-with-header',
+      component: html`<ia-sharing-options
+        identifier="goody"
+        type="book"
+        creator="Welsh, Charles"
+        description="The history of Little Goody Two-Shoes : otherwise called Mrs. Margery Two-Shoes ... [1766 edition]"
+        baseHost="www-shaner.arc,hive.org"
```
Comment (iisa) [inline demo/index.html]:
```suggestion
        baseHost="archive.org"
```

File: package.json (hunk 2 above)
Comment (iisa) [reply package.json]: i was going by this tag when i was cutting canary tags https://github.com/internetarchive/iaux-menu-slider/blob/master/package.json#L3

File: src/ia-sharing-options.js (hunk 3 above)
Comment (iisa) [reply src/ia-sharing-options.js:110]: they are :)
i am preserving the header for stand-alone usage

**Comment by iisa** [comment]
addressed review in-person with @shaneriley. will merge.

**Pull request merged**